### PR TITLE
Send locationAuthorization and locationAccuracyAuthorization to GET /config

### PR DESCRIPTION
**Pull request by nickpatrick**
Sends `locationAuthorization` (`DENIED`, `GRANTED_FOREGROUND`, `GRANTED_BACKGROUND`) and locationAccuracyAuthorization (`FULL`, `REDUCED`, or `null`) to GET /config. Stops sending `userId` and `deviceId` to `GET /config`.

Adds a `sessionId`, updated when the app is opened or foregrounded (with 5 minute timeout), also sent to `GET /config`.

Also updates project settings to Xcode 12, updates some sample code, and renames some functions for brevity.

### CODE REVIEW

**Comment by nickpatrick** [comment]
No need to review this week, just tagging for visibility.

File: RadarSDK/Radar.h [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 #import <CoreLocation/CoreLocation.h>
 #import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
```
Comment (pingxia0823) [inline RadarSDK/Radar.h:10]: Intentional?

File: RadarSDK/RadarUtils.m [hunk 2]
```diff
@@ -69,6 +69,35 @@ + (BOOL)locationBackgroundMode {
...
++ (NSString *)locationAccuracyAuthorization {
+    CLLocationManager *locationManager = [CLLocationManager new];
+    if (@available(iOS 14.0, *)) {
+        CLAccuracyAuthorization accuracyAuthorization = locationManager.accuracyAuthorization;
+        switch (accuracyAuthorization) {
+            case CLAccuracyAuthorizationReducedAccuracy:
+                return @"REDUCED";
+            case CLAccuracyAuthorizationFullAccuracy:
+                return @"FULL";
+            default:
```
Comment (pingxia0823) [inline RadarSDK/RadarUtils.m:93]: Will this default ever be reached?

File: RadarSDK/RadarUtils.h [hunk 3]
```diff
@@ -21,13 +21,15 @@ NS_ASSUME_NONNULL_BEGIN
 + (NSString *)deviceType;
 + (NSString *)deviceMake;
 + (BOOL)locationBackgroundMode;
++ (NSString *)locationAuthorization;
++ (NSString *)locationAccuracyAuthorization;
```
Comment (pingxia0823) [inline RadarSDK/RadarUtils.h:25]: This is nullable I think

File: RadarSDK/RadarSettings.m [hunk 4]
```diff
@@ -43,6 +45,23 @@ + (NSString *)installId {
     return installId;
 }
 
++ (NSString *)sessionId {
+    return [NSString stringWithFormat:@"%.f", [[NSUserDefaults standardUserDefaults] doubleForKey:kSessionId]];
+}
+
++ (BOOL)updateSessionId {
+    double timestampSeconds = [[NSDate date] timeIntervalSince1970];
+    double sessionIdSeconds = [[NSUserDefaults standardUserDefaults] doubleForKey:kSessionId];
```
Comment (pingxia0823) [inline RadarSDK/RadarSettings.m:54]: I kind of forget. If the key is not existent, will `sessionIdSeconds` be nil or 0?

File: RadarSDK/Radar.h (hunk 1 above)
Comment (pingxia0823) [reply RadarSDK/Radar.h:10]: I mean should we put it in the .h or .m file

File: RadarSDK/Radar.m [hunk 5]
```diff
@@ -28,10 +28,16 @@ + (id)sharedInstance {
 
 + (void)initializeWithPublishableKey:(NSString *)publishableKey {
     [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:@"Initializing"];
```
Comment (pingxia0823) [inline RadarSDK/Radar.m:30]: Shall we add a note to encourage the developers to call this method within their 'didLaunchWithOptions' method?

File: RadarSDK/RadarSettings.m [hunk 6]
```diff
@@ -43,6 +45,23 @@ + (NSString *)installId {
...
+    if (timestampSeconds - sessionIdSeconds > 300) {
```
Comment (pingxia0823) [inline RadarSDK/RadarSettings.m:55]: nice, so we guard against the quick / unintentional backgrounding and then foregrounding

**Review by pingxia0823 — COMMENTED** [review]
mostly nit comments

File: RadarSDK/Radar.m (hunk 5 above)
Comment (nickpatrick) [reply RadarSDK/Radar.m:30]: Yup, it's in the docs: https://radar.io/documentation/sdk/ios#initialize-sdk

File: RadarSDK/RadarSettings.m (hunk 6 above)
Comment (nickpatrick) [reply RadarSDK/RadarSettings.m:55]: Yep

File: RadarSDK/RadarSettings.m (hunk 4 above)
Comment (nickpatrick) [reply RadarSDK/RadarSettings.m:54]: 0

File: RadarSDK/RadarUtils.h (hunk 3 above)
Comment (nickpatrick) [reply RadarSDK/RadarUtils.h:25]: Good catch, will specify

File: RadarSDK/RadarUtils.m (hunk 2 above)
Comment (nickpatrick) [reply RadarSDK/RadarUtils.m:93]: It shouldn't, at least on the current version of iOS (not sure if Apple will add more enum values in the future)

File: RadarSDK/Radar.h (hunk 1 above)
Comment (nickpatrick) [reply RadarSDK/Radar.h:10]: Good point, added to listen for `UIApplicationWillEnterForegroundNotification` for `sessionId`, but can probably move it to the `.m` file

File: RadarSDK/RadarUtils.h (hunk 3 above)
Comment (nickpatrick) [reply RadarSDK/RadarUtils.h:25]: Changed this to return `FULL` pre-iOS 14, so never `nil`

**Pull request merged**